solr/core/src/java/org/apache/solr/cli/CreateTool.java

```diff
@@ -43,24 +68,295 @@ public String getName() {
+      }
+
+      FileUtils.copyDirectoryToDirectory(confDir.toFile(), coreInstanceDir.toFile());
+
+      echoIfVerbose(
+          "\nCopying configuration to new core instance directory:\n"
+              + coreInstanceDir.toAbsolutePath(),
+          cli);
+    }
+
+    echoIfVerbose("\nCreating new core '" + coreName + "' using CoreAdminRequest", cli);
+
+    try {
+      CoreAdminResponse res = CoreAdminRequest.createCore(coreName, coreName, solrClient);
+      if (cli.hasOption(SolrCLI.OPTION_VERBOSE.getOpt())) {
+        echo(res.jsonStr());
+        echo("\n");
       } else {
-        tool = new CreateCoreTool(stdout);
+        echo(String.format(Locale.ROOT, "\nCreated new core '%s'", coreName));
       }
-      tool.runImpl(cli);
+    } catch (Exception e) {
+      /* create-core failed, cleanup the copied configset before propagating the error. */
+      PathUtils.deleteDirectory(coreInstanceDir);
+      throw e;
+    }
+  }
+
+  protected void createCollection(CommandLine cli) throws Exception {
+    String zkHost = SolrCLI.getZkHost(cli);
+    try (CloudSolrClient cloudSolrClient =
+        new CloudHttp2SolrClient.Builder(Collections.singletonList(zkHost), Optional.empty())
+            .withInternalClientBuilder(
+                new Http2SolrClient.Builder()
+                    .withIdleTimeout(30, TimeUnit.SECONDS)
+                    .withConnectionTimeout(15, TimeUnit.SECONDS))
+            .build()) {
+      echoIfVerbose("Connecting to ZooKeeper at " + zkHost, cli);
+      cloudSolrClient.connect();
+      createCollection(cloudSolrClient, cli);
+    }
+  }
+
+  protected void createCollection(CloudSolrClient cloudSolrClient, CommandLine cli)
+      throws Exception {
+
+    String collectionName = cli.getOptionValue("name");
+    final String solrInstallDir = System.getProperty("solr.install.dir");
+    String confName = cli.getOptionValue("confname");
+    String confDir = cli.getOptionValue("confdir", "_default");
+    ensureConfDirExists(Paths.get(confDir), Paths.get(solrInstallDir));
+    printDefaultConfigsetWarningIfNecessary(cli);
+
+    Set<String> liveNodes = cloudSolrClient.getClusterState().getLiveNodes();
+    if (liveNodes.isEmpty())
+      throw new IllegalStateException(
+          "No live nodes found! Cannot create a collection until "
+              + "there is at least 1 live node in the cluster.");
+
+    String solrUrl = cli.getOptionValue("solrUrl");
+    if (solrUrl == null) {
+      String firstLiveNode = liveNodes.iterator().next();
+      solrUrl = ZkStateReader.from(cloudSolrClient).getBaseUrlForNodeName(firstLiveNode);
+    }
+
+    // build a URL to create the collection
+    int numShards = Integer.parseInt(cli.getOptionValue("shards", String.valueOf(1)));
+    int replicationFactor =
+        Integer.parseInt(cli.getOptionValue("replicationFactor", String.valueOf(1)));
+
+    final String configsetsDir = solrInstallDir + "/server/solr/configsets";
```

Comment:
Seeing string concatenation here vs. working with Path API.  Also, this looks redundant; with logic inside `getFullConfDir` suggesting that that method should maybe take configsetsDir instead?